sycl/include/syclcompat/memory.hpp

```diff
@@ -122,6 +122,44 @@ class pitched_data {
   size_t _pitch, _x, _y;
 };
 
+namespace experimental {
+#ifdef SYCL_EXT_ONEAPI_BINDLESS_IMAGES
+class image_mem_wrapper;
+namespace detail {
+static sycl::event dpct_memcpy(const image_mem_wrapper *src,
+                               const sycl::id<3> &src_id, pitched_data &dest,
+                               const sycl::id<3> &dest_id,
+                               const sycl::range<3> &copy_extend,
+                               sycl::queue q);
+static sycl::event
+dpct_memcpy(const pitched_data src, const sycl::id<3> &src_id,
+            image_mem_wrapper *dest, const sycl::id<3> &dest_id,
+            const sycl::range<3> &copy_extend, sycl::queue q);
+} // namespace detail
+#endif
+class image_matrix;
+namespace detail {
+static pitched_data to_pitched_data(image_matrix *image);
+}
+
+/// Memory copy parameters for 2D/3D memory data.
+struct memcpy_parameter {
+  struct data_wrapper {
+    pitched_data pitched{};
+    sycl::id<3> pos{};
+#ifdef SYCL_EXT_ONEAPI_BINDLESS_IMAGES
+    experimental::image_mem_wrapper *image_bindless{nullptr};
+#endif
+    image_matrix *image{nullptr};
+  };
+  data_wrapper from{};
+  data_wrapper to{};
+  sycl::range<3> size{};
+  syclcompat::detail::memcpy_direction direction{
```

Comment:
This exposes implementation details (`detail::memcpy_direction`). Can this be removed from the struct and be hidden like we previously did inside the [3D matrix memcpy](https://github.com/intel/llvm/blob/2e56a67eace630a6f877af222ff9239a045f753f/sycl/include/syclcompat/memory.hpp#L385)? 

I don't see how this `memcpy_direction` parameter is used nor passed around it the existing PR.